## Домашнее задание 7 ##
### Это задание выполнено Павлом Овчинниковым при обучении на курсе "Теория вероятностей и математическая статистика" ###
### Преподаватель Дамир Тлеубаев ###

## Задание 1)  
### Выбрать тест и проверить, есть ли различия между выборками: 
### Даны две  независимые выборки. Не соблюдается условие нормальности. Сделайте вывод по результатам, полученным с помощью функции:
### x1  380, 420, 290
### y1 140, 360, 200, 900 
### Решение: Исходя из условий независимости 2х выборок и отсутствия соблюдения нормальности применим непараметрический тест Манна-Уитни.

In [4]:
x1=np.array([380,420,290])
y1=np.array([140,360,200,900])
stats.mannwhitneyu(x1, y1)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

### Вывод: До уровня статистической значимости альфа около 0,6285 считаем, что между выборками различий нет. Т.е. при всех принятых альфа от 1% до 10% принимаем гипотезу H0 - различий нет.

______________________________________________________________________________
## Задание 2) 
### Выбрать тест и проверить, есть ли различия между выборками: Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?
### 1е измерение до приема препарата: 150, 160, 165, 145, 155
### 2е измерение через 10 минут: 140, 155, 150,  130, 135
### 3е измерение через 30 минут: 130, 130, 120, 130, 125
### Решение: Три выборки зависимы и это анализ повторных измерений, поэтому выбираем критерий Фридмана

In [5]:
x1=np.array([150, 160, 165, 145, 155])
x2=np.array([140, 155, 150, 130, 135])
x3=np.array([130, 130, 120, 130, 125])
stats.friedmanchisquare(x1, x2, x3)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

### Вывод: Т.к. p-value около 0,0083, то для всех уровней статистической значимости альфа  10%, 5%, и 1% есть статистические различия.

______________________________________________________________________________
## Задание 3) 
### В Задание 2 сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.
### Решение: Т.к. две выборки зависимы и это анализ по критерию Уилкоксона

In [6]:
import numpy as np
import scipy.stats as stats
x1=np.array([150, 160, 165, 145, 155])
x2=np.array([140, 155, 150, 130, 135])
stats.wilcoxon(x1, x2)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

### Вывод: Т.к. p-value = 0.0625, то для  уровней статистической значимости альфа 5% и 1% статистических различий нет, а для 10% есть.

______________________________________________________________________________
## Задание 4) 
### Даны 3 группы  учеников плавания.
### В 1 группе время на дистанцию 50 м составляют: 56, 60, 62, 55, 71, 67, 59, 58, 64, 67
### Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53
### Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54
### Решение: Т.к. три выборки независимы, то это анализ по критерию Крускала-Уолисса

In [9]:
x1=np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
x2=np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
x3=np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])
stats.kruskal(x1, x2, x3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

### Вывод: Т.к. p-value = 0.065, то для уровней статистической значимости альфа 5% и 1% статистических различий нет, а для 10% есть.

______________________________________________________________________________
## Задание 5) 
### Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%
### 2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

### Решение: выбираем двусторонний параметрический тест Стьюдента. Гипотеза Н0 это М=2,5 и Н1 это М не равно 2,5. Доверительная вероятность 0,95 и число степеней свободы 10-1 = 9

In [15]:
import scipy as sp

size = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
n = 10
mean = size.mean()
std = size.std(ddof=1)
tn = (mean-2.5)/(std/np.sqrt(n)) # tn- наблюдаемое
tk = sp.stats.t.ppf(q=1-.05/2,df=9) # t- критическое

mean, std, tn, tk

(2.5279999999999996,
 0.1572542173961923,
 0.5630613661802959,
 2.2281388519649385)

### Значение tn- наблюдаемое попадает в критическую область, таким образом, гипотезу H1 не отвергаем, партия изготавливается со средним арифметическим не равным 2,5 